src/Microsoft.Diagnostics.Monitoring.StartupHook/Exceptions/Pipeline/Steps/FilterMonitorExceptionPipelineStep.cs

```diff
@@ -0,0 +1,35 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System;
+
+namespace Microsoft.Diagnostics.Monitoring.StartupHook.Exceptions.Pipeline.Steps
+{
+    /// <summary>
+    /// An exception pipeline step that will only allow an exception to be
+    /// processed further if it was not caused by a dotnet-monitor in-proc feature.
+    /// </summary>
+    internal sealed class FilterMonitorExceptionPipelineStep
+    {
+        private readonly ExceptionPipelineDelegate _next;
+
+        public FilterMonitorExceptionPipelineStep(ExceptionPipelineDelegate next)
+        {
+            ArgumentNullException.ThrowIfNull(next);
+
+            _next = next;
+        }
+
+        public void Invoke(Exception exception, ExceptionPipelineExceptionContext context)
+        {
+            ArgumentNullException.ThrowIfNull(exception);
+
+            if (MonitorExecutionContextTracker.IsInMonitorContext())
```

Comment:
Does the invocation of AppDomain.Exception events happen completely synchronously? Is there any possibility that an exception thrown within a bounded region of our code will get a callback _after_ the Dispose call?

Reply:
Our exception handling is done synchronously so that scenario should never happen.

ref: 
> After the event has been raised and event handlers have been invoked, the common language runtime (CLR) begins to search for a handler for the exception.

https://learn.microsoft.com/en-us/dotnet/api/system.appdomain.firstchanceexception?view=net-8.0#remarks